CEPHSTOR-87 [WIP]  rgw docs initial draft

letterj opened the pull request:
This is the first draft at a document detailing instruction for setting up the Ceph Object Gateway in three different use cases

1.   Ceph rgw install 
2.   Ceph rgw install with a OpenStack Keystone only. (No other OpenStack components)
3.   Ceph rgw install with Openstack Keystone and It's used as a replacement for OpenStack Swift

To include:
Any rpc-maas bit?

review:  
variables in  https://github.com/rcbops/rpc-ceph/blob/master/playbooks/ceph-keystone-rgw.yml

andymcc commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+ceph-rgw03
+
+[keystone_all]
+keystone_host
```
to prevent the need for a host_vars file for that host, we could do
keystone_host ansible_host=<ip_address>

We only really use that address to delegate to it - but maybe a note that says:
"NB You will need to have access to the keystone host/container via ssh, from the deployment host" (that should already be in place, but worth noting it anyway.)

letterj replied:
Added the ansible_host=<ip_address> and the comment

andymcc commented on docs/rgw_setup.md:
````diff
@@ -0,0 +1,136 @@
+Make changes these lines from the file **playbooks/group_vars/rgws.yml** because they are used for integration with OpenStack
+```
+# Replace these 
+radosgw_keystone_service_name: "swift"
````
I think lets make this a patch to do this rather than asking them to manually do it.
If we change those group_vars/rgws.yml files to those values it seems to just make sense. Then we can remove this step.

letterj replied:
I don't know how changing them in an OpenStack deploy will affect other things.

andymcc replied:
We can test that, but description one at a minimum should make no difference. I'll put a PR in for this now, and if it passes our keystone-rgw tests it should be fine. Since no openstack services use the RadosGW (glance for images uses direct rbd access, as do cinder/nova) so it would only be swift client (which we test in our keystone_rgw gate).

letterj replied:
removed reference to changing service name

andymcc commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+* internal_lb_vip_address
+* service_region
+* radosgw_keystone **(set to True)**
+* keystone_service_adminurl
```
don't need keystone_service_adminurl if internal_lb_vip_address is specified.

letterj replied:
removed keystone_service_adminurl

andymcc commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+* radosgw_keystone_admin_user
+* radosgw_keystone_admin_password
+* radosgw_keystone_admin_tenant
+* What from this?   https://github.com/rcbops/rpc-ceph/blob/master/playbooks/ceph-keystone-rgw.yml
```
keystone_admin_user_name
keystone_auth_admin_password
keystone_admin_tenant_name
keystone_service_adminuri_insecure

NB These all need to match whatever was setup on the openstack side, so the password/admin_user/tenant
all have to match what is in the keystone/openstack deployment.

letterj replied:
Added variables to the list and the comment

andymcc commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+
+### To install rgw addtions to the inventory file.   
+
+Add the **rgw** group with a list of hosts into the ceph environment's inventory file. 
```
rgws

letterj replied:
Fixed

andymcc commented on docs/rgw_setup.md:
````diff
@@ -0,0 +1,136 @@
+Add the **rgw** group with a list of hosts into the ceph environment's inventory file. 
+For example:
+```bash
+[rgw]
````
[rgws]

letterj replied:
Fixed

andymcc commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+ceph-rgw03
+
+[keystone_all]
+keystone_host
```
same comment about the host address

letterj replied:
Fixed

andymcc commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+* radosgw_keystone_admin_user
+* radosgw_keystone_admin_password
+* radosgw_keystone_admin_tenant
+* What from this?   https://github.com/rcbops/rpc-ceph/blob/master/playbooks/ceph-keystone-rgw.yml
```
same as above.

letterj replied:
Fixed

andymcc commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+
+
+## Load Balancers and SSL Certs
+A load balancer should be deployed prior to ceph rgw install and you have VIP.  Currently RPC terminates SSL in the load balancer so the Ceph Object Store does not need anything for SSL set
```
If theyre using an f5 thats ok, if they're using haproxy we may need to explain that a bit more.
It's a bit trickier to setup the vip for ceph in haproxy because it isn't integrated in the RPC-O inventory, but I think we can get around that.

letterj commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+
+
+## Installation of rgw only
+To Install rgw **ONLY** and **NOT** use any of rpc-ceph configured varibles 
```
The first use case has nothing to do with KeyStone or OpenStack.

andymcc replied:
Agreed, but rpc-ceph has vars for ceph in general, this line implies we're not setting any vars at all. Also, since the vars will still work as long as you haven't set "keystone_radosgw: True" the deployer/user will still get the defaults that we set here, without any of the keystone integration.

letterj replied:
Sounds like you would rather we do away with the first use case all together.    That's cool.

letterj replied:
I removed the first use case

letterj commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+
+### To install rgw addtions to the inventory file.   
+
+Add the **rgw** group with a list of hosts into the ceph environment's inventory file. 
```
Fixed

letterj commented on docs/rgw_setup.md:
````diff
@@ -0,0 +1,136 @@
+Add the **rgw** group with a list of hosts into the ceph environment's inventory file. 
+For example:
+```bash
+[rgw]
````
Fixed

letterj commented on docs/rgw_setup.md:
````diff
@@ -0,0 +1,136 @@
+ceph-rgw03
+```
+
+### Installation variables
````
There quite a few assumptions made to the configuration in rgws.yml file.   The port is set to 8080, etc.     IMO the first use case is more for someone experimenting or testing but we can certainly remove it if you think it isn't necessary

andymcc replied:
The only default vars we set for radosgw, which are needed:
radosgw_address_block: "{{ public_network }}"
radosgw_civetweb_bind_ip: "{{ hostvars[inventory_hostname]['ansible_all_ipv4_addresses'] | ipaddr(radosgw_address_block) | first }}"
radosgw_civetweb_port: 8080

The port is the default value for ceph anyway, so if you don't specify it you should still get 8080 (or it will fail because its not specified), the other 2 are required to tell radosgw where to listen, regardless of keystone integration or not. 
All the other vars in rgws.yml won't be used unless you set radosgw_keystone: True, or you are using some old version of ceph-ansible which isn't shipped in rpc-ceph.

letterj replied:
I completely removed the first use case to

letterj commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+
+
+## Installation of rgw with Stand Alone Openstack KeyStone
+To Install rgw and connect it to a Stand Alone instance of OpenStack Keystone.   Please refer to other documentation for the RPC guidelines for installing and setting up OpenStack Keystone.   This ceph rgw guide assumes you have an working OpenStack Keystone system in place
```
reworded the sentence

letterj commented on docs/rgw_setup.md:
```diff
@@ -0,0 +1,136 @@
+
+### To install rgw addtions to the inventory file.   
+
+Add the **rgw** group with a list of hosts into the ceph environment's inventory file. 
```
Fixed

letterj commented on docs/rgw_setup.md:
````diff
@@ -0,0 +1,136 @@
+Add the **rgw** group with a list of hosts into the ceph environment's inventory file. 
+For example:
+```bash
+[rgw]
````
Fixed

letterj commented:
recheck_cit_xenial_functional_test

letterj commented:
Based on further investigation and discussion I removed the second use case since there was basically no difference between the use cases of ceph rgw and OpenStack Keystone with or without OpenStack other.  

I also reworked the first use case to make it ceph rgw installed without an Auth System

prometheanfire commented:
It looks like you are trying to run openstack code (to set up the endpoints).  Why does rpc-ceph try to talk to openstack?  I should be able to run this without communicating to openstack at all.

letterj commented:
The communication with OpenStack of for the KeyStone integration.    If you don't want to run the playbook inside rpc-ceph to set up KeyStone endpoints you can comment it out and set up the endpoints manually.

letterj commented:
Oh course you have to EXPECT access to OSA variable in a production deployment .  If now that is the purpose of using ceph-ansible in the first place?

prometheanfire commented:
ceph could be managed separately from the openstack environment, meaning no direct access to the vars.  Meaning they have to be copied over and inserted into rpc-ceph's vars file.

letterj commented:
I'm confused:   Isn't this copying them over to a "vars" file?  
+### Make sure you have each of these variables set in your vars file
 +* public_network
 +* internal_lb_vip_address
 +* service_region
 +* radosgw_keystone **(set to True)**
 +* radosgw_keystone_admin_user
 +* radosgw_keystone_admin_password
 +* radosgw_keystone_admin_tenant
 +These all need to match whatever was setup on the openstack side, so the password/admin_user/tenant
 +all have to match what is in the keystone/openstack deployment.
 +* keystone_admin_user_name
 +* keystone_auth_admin_password
 +* keystone_admin_tenant_name
 +* keystone_service_adminuri_insecure

Exactly where do you want this vars file to live?

andymcc commented:
I'm not sure I understand your issue @prometheanfire - it needs to know about the vars in order to setup the radosgw keystone integration. So you need to manually copy those over (or automate them in some fashion from the rpc-o side) - there isn't a "running of openstack" or anything openstack related in that step.

prometheanfire commented:
ya, I think I misread a portion of this (or conflated it with the test stuff earlier), dismissing the review.

prometheanfire commented:
merged a WIP PR?

andymcc commented:
The commits aren't listed as WIP, just the initial pr and the title didn't
change.

On 25 Jan 2018 17:59, "Matthew Thode" <notifications@github.com> wrote:

> merged a WIP PR?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/rcbops/rpc-ceph/pull/63#issuecomment-360548162>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABb__tSf4tXDi9Gkt1HEoGxoOfkvsNddks5tOMENgaJpZM4Rrg6Z>
> .
>

prometheanfire commented:
ah, that's better then, wish we used gerrit :P
